server/pkg/provision/auth.go

```diff
@@ -43,3 +51,121 @@ func emailSettings(ctx context.Context, s store.Storer) error {
 		})
 	})
 }
+
+// addAuthSuperUsers assigns BYPASS roles to users from AUTH_PROVISION_SUPER_USER value
+// When in Production, Corteza should stop and report an error.
+func addAuthSuperUsers(ctx context.Context, log *zap.Logger, s store.Storer, authOpt options.AuthOpt) (err error) {
+	var (
+		envOpt = options.Environment()
+	)
+
+	if authOpt.ProvisionSuperUser == "" {
+		return nil
+	}
+
+	if envOpt.IsProduction() {
+		log.Warn(fmt.Sprint("when in production environment (ENVIRONMENT=production) you cannot provision" +
+			"super users; set the environment to dev (ENVIRONMENT=dev) to provision super users"))
+		return
+	}
+
+	users := strings.Split(authOpt.ProvisionSuperUser, ";")
+
+	for _, usr := range users {
+		u := prepareUser(usr)
+
+		//check if the email address is valid
+		if !mail.IsValidAddress(u.Email) {
+			log.Warn(fmt.Sprintf("Email address %s is invalid", u.Email))
+			continue
+		}
+
+		// skip existing email
+		_, err = s.LookupUserByEmail(ctx, u.Email)
+		if err != store.ErrNotFound {
+			log.Warn(fmt.Sprintf("Email address already %s exists", u.Email))
+			continue
+		}
+
+		// skip existing handle
+		_, err = s.LookupUserByHandle(ctx, u.Handle)
+		if err != store.ErrNotFound {
+			log.Warn(fmt.Sprintf("Handle %s already exists", u.Handle))
+			continue
+		}
+
+		err = store.Tx(ctx, s, func(ctx context.Context, s store.Storer) (err error) {
+			if err = store.CreateUser(ctx, s, u); err != nil {
+				return err
+			}
+
+			if err = service.SetPasswordCredentials(ctx, s, u.ID, u.Email); err != nil {
+				return err
+			}
+
+			log.Warn(fmt.Sprintf("User %#v created", u))
+
+			//assign the user a bypass role
+			for _, r := range internalAuth.BypassRoles() {
+				m := &types.RoleMember{UserID: u.ID, RoleID: r.ID}
+				if err = store.CreateRoleMember(ctx, s, m); err != nil {
+					return err
+				}
+			}
+
+			return
+		})
+
+		if err != nil {
+			return err
+		}
+	}
+
+	return
+}
+
+// prepareUser creates and fills a new user depending on the number of arguments provided
+func prepareUser(user string) *types.User {
+	var (
+		nextID = func() uint64 {
+			return id.Next()
+		}
+	)
```

Comment:
Isn't there already a `nextID` func in this package?
If not, I think we should have it defined globally instead of here (there isn't any harm in doing that)